.github/workflows/robot/internal/leave/leave_test.go

```diff
@@ -0,0 +1,254 @@
+			desc:       "one-business-day before-leave--omit",
+		},
+		{
+			currentDay: time.Date(2021, time.Month(1), 1, 0, 0, 0, 0, time.UTC),
+			expected:   true,
+			desc:       "first-day-of-leave--omit",
+		},
+		{
+			currentDay: time.Date(2021, time.Month(1), 4, 0, 0, 0, 0, time.UTC),
+			expected:   true,
+			desc:       "during-leave--omit",
+		},
+		{
+			currentDay: time.Date(2021, time.Month(1), 5, 0, 0, 0, 0, time.UTC),
+			expected:   true,
+			desc:       "last-day-of-leave--omit",
+		},
+		{
+			currentDay: time.Date(2021, time.Month(1), 6, 0, 0, 0, 0, time.UTC),
+			expected:   true,
+			desc:       "one-business-day-after-leave--omit",
+		},
+		{
+			currentDay: time.Date(2021, time.Month(1), 7, 0, 0, 0, 0, time.UTC),
+			expected:   false,
+			desc:       "two-business-days-after-leave--don't-omit",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			res := req.shouldOmitEmployee(test.currentDay)
+			require.Equal(t, test.expected, res)
+
+		})
+	}
+}
+
+func TestShouldOmitRangeOverOnlyBusinessDays(t *testing.T) {
+	// Leave is a business week.
+	req := employeeLeaveRequest{
+		StartDate: Time{time.Date(2021, time.Month(1), 18, 0, 0, 0, 0, time.UTC)},
+		EndDate:   Time{time.Date(2021, time.Month(1), 22, 0, 0, 0, 0, time.UTC)},
+	}
+
+	tests := []struct {
+		currentDay time.Time
+		expected   bool
+		desc       string
+	}{
+		{
+			currentDay: time.Date(2021, time.Month(1), 13, 0, 0, 0, 0, time.UTC),
+			expected:   false,
+			desc:       "three-business-days-before-leave--don't-omit",
+		},
+		{
+			currentDay: time.Date(2021, time.Month(1), 14, 0, 0, 0, 0, time.UTC),
+			expected:   true,
+			desc:       "two-business-days-before-leave--omit",
+		},
+		{
+			currentDay: time.Date(2021, time.Month(1), 15, 0, 0, 0, 0, time.UTC),
+			expected:   true,
+			desc:       "one-business-day-before-leave-before-weekend--omit",
+		},
+		{
+			currentDay: time.Date(2021, time.Month(1), 25, 0, 0, 0, 0, time.UTC),
+			expected:   true,
+			desc:       "one-business-day-after-leave-after-weekend--omit",
+		},
+		{
+			currentDay: time.Date(2021, time.Month(1), 26, 0, 0, 0, 0, time.UTC),
+			expected:   false,
+			desc:       "two-business-days-after-leave--don't-omit",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			res := req.shouldOmitEmployee(test.currentDay)
+			require.Equal(t, test.expected, res)
+		})
+	}
+}
+
+func TestShouldOmitTwoDaysOfLeave(t *testing.T) {
+	// Leave is not more than two business days.
+	req := employeeLeaveRequest{
+		StartDate: Time{time.Date(2021, time.Month(1), 18, 0, 0, 0, 0, time.UTC)},
+		EndDate:   Time{time.Date(2021, time.Month(1), 19, 0, 0, 0, 0, time.UTC)},
+	}
+
+	tests := []struct {
+		currentDay time.Time
+		expected   bool
+		desc       string
+	}{
+		{
+			currentDay: time.Date(2021, time.Month(1), 15, 0, 0, 0, 0, time.UTC),
+			expected:   false,
+			desc:       "three-business-days-before-two-day-leave--don't-omit",
```

Comment:
Jan 18, 2021 was a Monday so this is `"one-business-day-before-two-day-leave--don't-omit"`